pkg/sql/sem/tree/stmt.go

```diff
@@ -460,6 +460,15 @@ func (*AlterTenantReplication) StatementTag() string { return "ALTER TENANT REPL
 
 func (*AlterTenantReplication) cclOnlyStatement() {}
 
+// StatementReturnType implements the Statement interface.
+func (*AlterTenantRename) StatementReturnType() StatementReturnType { return Rows }
+
+// StatementType implements the Statement interface.
+func (*AlterTenantRename) StatementType() StatementType { return TypeDML }
```

Comment:
Since this is isn't a TypeDDL, I think that we may also want to update CanWriteData to ensure that this can't be run in read-only transactions.